cmake/Modules/FindGCSSDK_EP.cmake

```diff
@@ -0,0 +1,165 @@
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+# This module finds the GCS C++ SDK, installing it with an ExternalProject if
+# necessary. It then defines the imported targets GCSSDK::<component>, e.g.
+# GCSSDK::storage-client or GCSSDK::google_cloud_cpp_common.
+
+# Include some common helper functions.
+include(TileDBCommon)
+
+# If the EP was built, it will install the storage_client-config.cmake file,
+# which we can use with find_package. CMake uses CMAKE_PREFIX_PATH to locate find
+# modules.
+set(CMAKE_PREFIX_PATH ${CMAKE_PREFIX_PATH} "${TILEDB_EP_INSTALL_PREFIX}")
+
+# Try searching for the SDK in the EP prefix.
+set(GCSSDK_DIR "${TILEDB_EP_INSTALL_PREFIX}")
+
+# First check for a static version in the EP prefix.
+find_library(GCSSDK_LIBRARY
+  NAMES
+    libstorage_client${CMAKE_STATIC_LIBRARY_SUFFIX}
+  PATHS ${TILEDB_EP_INSTALL_PREFIX}
+  PATH_SUFFIXES lib
+  NO_DEFAULT_PATH
+)
+
+if (GCSSDK_LIBRARY)
+  set(GCSSDK_STATIC_EP_FOUND TRUE)
+  find_path(GCSSDK_INCLUDE_DIR
+    NAMES client.h
+    PATHS ${TILEDB_EP_INSTALL_PREFIX}
+    PATH_SUFFIXES include/google/cloud/storage
+    NO_DEFAULT_PATH
+  )
+else()
+  set(GCSSDK_STATIC_EP_FOUND FALSE)
+  # Static EP not found, search in system paths.
+  find_library(GCSSDK_LIBRARY
+    NAMES
+      libstorage_client
+    PATH_SUFFIXES lib bin
+    ${TILEDB_DEPS_NO_DEFAULT_PATH}
+  )
+  find_path(GCSSDK_INCLUDE_DIR
+    NAMES client.h
+    PATH_SUFFIXES include/google/cloud/storage
+    ${TILEDB_DEPS_NO_DEFAULT_PATH}
+  )
+endif()
+
+include(FindPackageHandleStandardArgs)
+FIND_PACKAGE_HANDLE_STANDARD_ARGS(GCSSDK
+  REQUIRED_VARS GCSSDK_LIBRARY GCSSDK_INCLUDE_DIR
+)
+
+if (NOT GCSSDK_FOUND)
+  if (TILEDB_SUPERBUILD)
+    message(STATUS "Could NOT find GCSSDK")
+    message(STATUS "Adding GCSSDK as an external project")
+
+    set(DEPENDS)
+    if (TARGET ep_curl)
+      list(APPEND DEPENDS ep_curl)
+    endif()
+    if (TARGET ep_openssl)
+      list(APPEND DEPENDS ep_openssl)
+    endif()
+    if (TARGET ep_zlib)
+      list(APPEND DEPENDS ep_zlib)
+    endif()
+
+    # Fetch the number of CPUs on the this sytem.
+    include(ProcessorCount)
+    processorcount(NCPU)
+
+    ExternalProject_Add(ep_gcssdk
+      PREFIX "externals"
+      URL "https://github.com/googleapis/google-cloud-cpp/archive/v0.20.0.zip"
+      URL_HASH SHA1=6e7931a0e62779dfbd07427373373bf1ad6f8686
+      BUILD_IN_SOURCE 1
+      PATCH_COMMAND
+        patch -N -p1 < ${TILEDB_CMAKE_INPUTS_DIR}/patches/ep_gcssdk/build.patch &&
+        patch -N -p1 < ${TILEDB_CMAKE_INPUTS_DIR}/patches/ep_gcssdk/ls.patch
+      CONFIGURE_COMMAND
+        cmake -Hsuper -Bcmake-out
+        -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}
+        -DBUILD_SHARED_LIBS=OFF
+        -DBUILD_SAMPLES=OFF
+        -DCMAKE_PREFIX_PATH=${TILEDB_EP_INSTALL_PREFIX}
+        -DOPENSSL_ROOT_DIR=${TILEDB_OPENSSL_DIR}
+      BUILD_COMMAND cmake --build cmake-out -- -j${NCPU}
```

Comment:
Same as above with configure, what is the need to specify the build command here, and if so let's use `${CMAKE_COMMAND}` like @ihnorton suggested